app/models/audit_activity/risk_assessment/risk_assessment_added.rb

```diff
@@ -4,46 +4,121 @@ def self.from(*)
-  def title(*)
-    "Risk assessment"
+  # Temporarily override getter until data is migrated
+  # TODO: Remove method once data migrated
+  def metadata
+    migrate_metadata_structure
   end
 
   def subtitle_slug
     "Added"
   end
 
+  def risk_assessment
+    RiskAssessment.find(metadata["risk_assessment"]["id"])
+  end
+
+  # Some older activities cannot refer to the file as it was not recorded on the activity and may have subsequently been deleted, or the filename recorded on the updated activity was ambiguous
+  def risk_assessment_file
+    ActiveStorage::Blob.find(metadata["risk_assessment"]["risk_assessment_file"]["id"]) if metadata["risk_assessment"]["risk_assessment_file"]
+  end
+
   def products_assessed
-    Product.find(metadata["product_ids"])
+    Product.find(metadata["risk_assessment"]["product_ids"])
   end
 
   def further_details
-    metadata["details"].presence
+    metadata["risk_assessment"]["details"]
   end
 
-  def assessed_by_name
-    if metadata["assessed_by_team_id"]
-      Team.find(metadata["assessed_by_team_id"])&.name
-    elsif metadata["assessed_by_business_id"]
-      Business.find(metadata["assessed_by_business_id"])&.trading_name
-    else
-      metadata["assessed_by_other"]
-    end
+  def risk_level
+    metadata["risk_assessment"]["risk_level"]
+  end
+
+  def custom_risk_level
+    metadata["risk_assessment"]["custom_risk_level"]
+  end
+
+  def assessed_on
+    Date.parse(metadata["risk_assessment"]["assessed_on"])
+  end
+
+  def assessed_by_team
+    Team.find_by(id: metadata["risk_assessment"]["assessed_by_team_id"])
   end
 
+  def assessed_by_business
+    Business.find_by(id: metadata["risk_assessment"]["assessed_by_business_id"])
+  end
+
+private
+
   # Do not send investigation_updated mail when test result updated. This
   # overrides inherited functionality in the Activity model :(
   def notify_relevant_users; end
+
+  # Migrates metadata from old structure to current. This method can be deleted once data is migrated.
+  # TODO: Remove method once data migrated
+  #
+  # Old structure:
+  #
+  # {
+  #   risk_assessment_id: risk_assessment.id,
+  #   assessed_on: risk_assessment.assessed_on,
+  #   risk_level: risk_assessment.risk_level,
+  #   custom_risk_level: risk_assessment.custom_risk_level,
+  #   assessed_by_team_id: risk_assessment.assessed_by_team_id,
+  #   assessed_by_business_id: risk_assessment.assessed_by_business_id,
+  #   assessed_by_other: risk_assessment.assessed_by_other,
+  #   details: risk_assessment.details,
+  #   product_ids: risk_assessment.product_ids
+  # }
+  def migrate_metadata_structure
+    metadata = self[:metadata]
+
+    # Already in new format
+    return metadata if metadata["risk_assessment"].present?
+
+    risk_assessment = RiskAssessment.find(metadata["risk_assessment_id"])
+    risk_assessment_updated = AuditActivity::RiskAssessment::RiskAssessmentUpdated.where("metadata->'risk_assessment_id' = ?", risk_assessment.id.to_s).count.positive?
```

Comment:
```suggestion
    risk_assessment_updated = AuditActivity::RiskAssessment::RiskAssessmentUpdated.where("metadata->'risk_assessment_id' = ?", risk_assessment.id.to_s).any?
```
I think this should work? not certain though as this queries the JSON field. But feel free to ignore.